src/scripts/main.js

```diff
@@ -1,3 +1,59 @@
 'use strict';
 
-// write code here
+// document.addEventListener('DOMContentLoaded', function() {
+//   const table = document.querySelector('table');
+//   const headers = table.querySelectorAll('th');
+
+//   headers.forEach(function(header) {
+//     header.addEventListener('click', function() {
+//       const column = this.cellIndex;
+//       const rows = Array.from(table.querySelectorAll('tbody tr'));
+
+//       rows.sort(function(rowA, rowB) {
+//         const cellA = rowA.cells[column].textContent.trim();
+//         const cellB = rowB.cells[column].textContent.trim();
+
+//         if (cellA < cellB) {
+//           return -1;
+//         } else if (cellA > cellB) {
+//           return 1;
+//         } else {
+//           return 0;
+//         }
+//       });
+
+//       table.querySelector('tbody').append(...rows);
+//     });
+//   });
+// });
+document.addEventListener('DOMContentLoaded', function() {
+  const table = document.querySelector('table');
+  const headers = table.querySelectorAll('th');
+  const sortDirections = Array.from(headers).fill(0);
+
+  headers.forEach(function(header, columnIndex) {
+    header.addEventListener('click', function() {
+      const column = this.cellIndex;
+      const rows = Array.from(table.querySelectorAll('tbody tr'));
+
+      rows.sort(function(rowA, rowB) {
+        const cellA = rowA.cells[column].textContent.trim()
+          .replace(/[$,]/g, '');
+        const cellB = rowB.cells[column].textContent.trim()
+          .replace(/[$,]/g, '');
+
+        return parseFloat(cellA) - parseFloat(cellB);
```

Comment:
Consider using the localeCompare() method with the 'numeric' option when comparing string values, to properly handle strings containing numbers.